models/classes/Cache/CacheItemPool.php

```diff
@@ -0,0 +1,77 @@
+<?php /** @noinspection ALL */
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+
+namespace oat\taoLti\models\classes\Cache;
+
+use common_exception_NoImplementation;
+use oat\oatbox\service\ConfigurableService;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+class CacheItemPool extends ConfigurableService implements CacheItemPoolInterface
```

Comment:
How is this working if all methods throw exception?

Reply:
This is used when we are saving access tokens. At this point, I'm generating new token each time.

Reply:
To you are actually not caching, right? I think we should implement this class in the complete way.

Reply:
Yes, this is right. I will will do that in actual implementation